pylint/extensions/code_style.py

```diff
@@ -69,6 +69,17 @@ class CodeStyleChecker(BaseChecker):
                 "default_enabled": False,
             },
         ),
+        "R6105": (
+            "Prefer typing.NamedTuple over collections.namedtuple",
+            "prefer-typing-namedtuple",
+            "Emitted when a call to collections.namedtuple() is detected. "
+            "This can be rewritten more elegantly as a class with type annotations. "
```

Comment:
```suggestion
            "'typing.NamedTuple' permits to add type hints, and use the "
            "well known 'class' keyword (it's also faster as it does not "
            "require an internal exec call)."
```
See [typing.NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple) for a way to add type hints for named tuples. It also provides an elegant notation using the [class](https://docs.python.org/3/reference/compound_stmts.html#class) keyword: